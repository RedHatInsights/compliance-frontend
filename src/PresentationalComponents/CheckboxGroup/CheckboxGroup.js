import React from 'react';
import { Field } from 'redux-form';
import propTypes from 'prop-types';
import { Checkbox } from '@patternfly/react-core';

const CheckboxFieldArray = ({ input, options }) => {
    const { name, onChange } = input;
    const inputValue = input.value;

    const handleChange = (checked, value) => {
        const arr = [...new Set(inputValue)];
        if (checked) {
            arr.push(value);
        } else {
            arr.splice(arr.indexOf(value), 1);
        }

        return onChange(arr);
    };

    return (
        <React.Fragment>
            {
                options.map(({ label, value, defaultChecked }, index) => (
                    <Checkbox key={value} id={value} label={label}
                        name={`${name}[${index}]`} value={value}
                        isChecked={inputValue === '' ? defaultChecked : inputValue.includes(value)}
                        onChange={(checked) => handleChange(checked, value)}
                    />
                ))
            }
        </React.Fragment>
    );
};

CheckboxFieldArray.propTypes = {
    input: propTypes.shape({
        name: propTypes.string.isRequired,
        onChange: propTypes.func,
        value: propTypes.array
    }),
    options: propTypes.arrayOf(propTypes.shape({
        label: propTypes.string.isRequired,
        value: propTypes.string.isRequired,
        defaultChecked: propTypes.bool
    })).isRequired
};

const CheckboxGroup = (props) => (
    <Field {...props} type="checkbox" component={CheckboxFieldArray} />
);

CheckboxGroup.propTypes = {
    options: propTypes.arrayOf(propTypes.shape({
        label: propTypes.string.isRequired,
        value: propTypes.string.isRequired,
        defaultChecked: propTypes.bool
    })).isRequired
};

export { CheckboxFieldArray };
export default CheckboxGroup;
