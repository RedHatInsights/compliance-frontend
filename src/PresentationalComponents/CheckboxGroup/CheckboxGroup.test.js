import { CheckboxFieldArray } from './CheckboxGroup';

describe('CheckboxFieldArray', () => {
  let input;
  let options;

  beforeEach(() => {
    input = { name: 'policies', value: '' };
    options = [
      {
        label: 'pci',
        value: '12345',
        defaultChecked: false,
      },
      {
        label: 'hipaa',
        value: '1111',
        defaultChecked: true,
      },
    ];
  });

  it('expect to render with default checked fields', () => {
    const component = shallow(
      <CheckboxFieldArray input={input} options={options} />
    );

    expect(toJson(component)).toMatchSnapshot();
  });

  it('expect to render without default checked fields', () => {
    options[0].defaultChecked = undefined;
    options[1].defaultChecked = undefined;

    const component = shallow(
      <CheckboxFieldArray input={input} options={options} />
    );

    expect(toJson(component)).toMatchSnapshot();
  });
});
