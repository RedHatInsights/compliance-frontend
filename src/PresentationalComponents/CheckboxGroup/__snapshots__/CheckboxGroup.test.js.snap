// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CheckboxFieldArray expect to render with default checked fields 1`] = `
<Fragment>
  <Checkbox
    className=""
    id="12345"
    isChecked={false}
    isDisabled={false}
    isValid={true}
    key="12345"
    label="pci"
    name="policies[0]"
    onChange={[Function]}
    value="12345"
  />
  <Checkbox
    className=""
    id="1111"
    isChecked={true}
    isDisabled={false}
    isValid={true}
    key="1111"
    label="hipaa"
    name="policies[1]"
    onChange={[Function]}
    value="1111"
  />
</Fragment>
`;

exports[`CheckboxFieldArray expect to render without default checked fields 1`] = `
<Fragment>
  <Checkbox
    className=""
    id="12345"
    isChecked={false}
    isDisabled={false}
    isValid={true}
    key="12345"
    label="pci"
    name="policies[0]"
    onChange={[Function]}
    value="12345"
  />
  <Checkbox
    className=""
    id="1111"
    isChecked={false}
    isDisabled={false}
    isValid={true}
    key="1111"
    label="hipaa"
    name="policies[1]"
    onChange={[Function]}
    value="1111"
  />
</Fragment>
`;
